### PR TITLE
Fix/recipe converted

### PR DESCRIPTION
**Pull request by vsoch**

### CODE REVIEW

File: spython/main/parse/docker.py [hunk 1]
```diff
@@ -65,7 +65,11 @@ def _from(self, line):
            line: the line from the recipe file to parse for FROM
 
         '''
-        self.fromHeader = self._setup('FROM', line)
+        fromHeader = self._setup('FROM', line)
+
+        # Singularity does not support AS level
+        self.fromHeader = re.sub("AS level", "", fromHeader, flags=re.I)
```
Comment (samcmill) [inline spython/main/parse/docker.py]: An unfortunate autocorrect in the original issue report.  This was intended to be 'AS devel', but the name can be anything.  So the regex should be something like "AS \S+"...

File: spython/main/parse/environment.py [hunk 2]
```diff
@@ -36,14 +36,13 @@ def parse_env(envlist):
...
-    # Detail work not yet done here, see link above for possible cases
-
     for env in envlist:
 
-        pieces = re.split("( |\\\".*?\\\"|'.*?')", env)
-        pieces = [p for p in pieces if p.strip()]
-        export = "=".join(pieces)
-        exports.append(export)
+        if "=" in env:
+            exports = exports + re.split(' ', env)
```
Comment (samcmill) [inline spython/main/parse/environment.py]: will this handle the case `ENV A=B X="Y Y Y"`?

Comment (vsoch) [reply spython/main/parse/environment.py]: I don't think that is valid, at least per their docs. If you use the second format, the equals must be present I think? See https://docs.docker.com/engine/reference/builder/#env


> The second form, ENV <key>=<value> ..., allows for multiple variables to be set at one time. Notice that the second form uses the equals sign (=) in the syntax, while the first form does not. Like command line parsing, quotes and backslashes can be used to include spaces within values.

File: spython/main/parse/docker.py (hunk 1 above)
Comment (vsoch) [reply spython/main/parse/docker.py]: ah my mistake! I should have looked at the docs first!

**Comment by vsoch** [comment]
okay this is super fugly, but give it a try :)

```python
def parse_env(envlist):
    '''parse_env will parse a single line (with prefix like ENV removed) to
    a list of commands in the format KEY=VALUE For example:

        ENV PYTHONBUFFER 1 --> [PYTHONBUFFER=1]

    ::Notes
      Docker: https://docs.docker.com/engine/reference/builder/#env

    '''
    if not isinstance(envlist, list):
        envlist = [envlist]

    exports = [] 

    for env in envlist:
        pieces = re.split("( |\\\".*?\\\"|'.*?')", env)
        pieces = [p for p in pieces if p.strip()]

        while len(pieces) > 0:
            current = pieces.pop(0)
            if current.endswith('='):
                next = pieces.pop(0)
                exports.append("%s%s" %(current, next))
            elif '=' in current:
                exports.append(current)
            else:
                next = pieces.pop(0)
                exports.append("%s=%s" %(current, next))

    return exports
```

File: spython/main/parse/environment.py (hunk 2 above)
Comment (vsoch) [reply spython/main/parse/environment.py]: see my function below! It's not pretty, but possibly would meet all cases.

**Comment by samcmill** [comment]
Looks good!

**Comment by vsoch** [comment]
Awesome! Merging. I'll rebuild the image and then test the other issue.

**Pull request merged**